lib/rules/no-invalid-position-at-import-rule/index.js

```diff
@@ -0,0 +1,50 @@
+// @ts-nocheck
+
+'use strict';
+
+const report = require('../../utils/report');
+const ruleMessages = require('../../utils/ruleMessages');
+const validateOptions = require('../../utils/validateOptions');
+
+const ruleName = 'no-invalid-position-at-import-rule';
+
+const messages = ruleMessages(ruleName, {
+	rejected: 'Unexpected invalid position @import rule',
+});
+
+function rule(actual) {
+	return (root, result) => {
+		const validOptions = validateOptions(result, ruleName, { actual });
+
+		if (!validOptions) {
+			return;
+		}
+
+		let invalidPosition = false;
+
+		root.walk((node) => {
+			if (node.type === 'comment' || (node.type === 'atrule' && /^charset$/i.test(node.name))) {
+				return;
+			}
+
+			if (node.type === 'atrule' && /^import$/i.test(node.name)) {
```

Comment:
[IMO] It seems a bit readable to me if using `toLowerCase`. But this is just my opinion, so you can ignore it. 😃 

```suggestion
			const nodeName = node.name.toLowerCase();

			if (node.type === 'comment' || (node.type === 'atrule' && nodeName === 'charset')) {
				return;
			}

			if (node.type === 'atrule' && nodeName === 'import') {
```

Reply:
Suggestion looks suitable. Just made some research and `toLowerCase()` used much more often in rules.
@jeddy3 what do you think about it? Do we have any convention on this or should create one if not?

Reply:
I had the same thought about `toLowerCase` as @ybiquitous when I was first reviewing the pull request as we've used that approach in loads of places. However, the regex approach sort of grew on me because we use it for filtering at-rules and declarations when walking (like we did in the [last rule we added](https://github.com/stylelint/stylelint/pull/5167/files#diff-280597139af609e4ef602c8171d06774830a2e1e98e2bc7536a1203c73abe2caR30)), so I didn't mention it.

> Do we have any convention on this or should create one if not?

I'm all for conventions, especially within rules, because they're such a time saver when reviewing code.

I'm not sure which of the approaches is more readable, though. So, let's go with @ybiquitous's `toLowerCase` suggestion for now to be consistent with how we're treating names and values in other rules. We can then, if we want, refactor across the board later down the line.

Reply:
Following on, we'll probably stick with `toLowerCase` as the convention because (having looked myself) we often:
- do more string manipulation at the same time, e.g. trimming and slicing
- look for matches in arrays and sets